### DISCUSSION TITLE
The execution of doris histogram with a large amount of data seems to be very easy to OOM. Is my understanding correct?

### DISCUSSION THREAD

**Discussion by suimenno3002** [apache/doris#30916] [A - General / Q&A]
It seems that map<value, cnt> is constructed in memory, and merge also needs to send this thing. The final converted equal weight histogram, the last step must be completed on an instance? In this case, the amount of data is too large to be stored in the memory, and this will be OOM.

**Comment by LemonLiTree** [comment]
which version？

**Reply by suimenno3002** [reply]
I have not actually observed this problem. This question came to me when I read the Doris source code.
